pyfar/dsp/dsp_interpolation.py

```diff
@@ -0,0 +1,734 @@
+import numpy as np
+from scipy.special import iv as bessel_first_mod
+from scipy.interpolate import interp1d
+import scipy.signal as sgn
+import matplotlib.pyplot as plt
+import pyfar as pf
+
+
+def smooth_fractional_octave(signal, width, mode="magnitude",
+                             window="boxcar"):
+    """
+    Smooth spectrum with a fractional octave width.
+
+    The smoothing is done according to Tylka et al. 2017 [#]_ (method 2) in
+    three steps:
+
+    1. Interpolate the spectrum to a logarithmically spaced frequency scale
+    2. Smooth the spectrum by convolution with a smoothing window
+    3. Interpolate the spectrum to the original linear frequency scale
+
+
+    Parameters
+    ----------
+    signal : pyfar.Signal
+        The input data.
+    width : number
+        The width of the smoothing window in octaves, e.g., 1/3 will apply
+        third octave smoothing.
+    mode : str, optional
+
+        ``"magnitude"``
```

Comment:
@sikersten pointed out that it might be good to be consistent with InterpolateSpectrum, where the mode `magnitude` is used in the same sense. I would suggest to leave this as it is, but find a better name for `magnitude_copy` below.

Reply:
I'd say that's a bit of a different case. In the interpolation it implies, that the magnitude is interpolated, which means, that you don't get a new result for the phase. Here I'd still argue, that if you smooth one, the other remains untouched, you don't generate new data, but you modify the present data. So, I'd still argue, that consistency isn't the biggest issue here.

Reply:
I agree that there are differences, but they might not be apparent to everyone. But I don't have a strong opinion for the names. What would you suggest?

Reply:
I'm just wondering now: The current magnitude option could easily be replaced by passing a zero-phase signal and smooth only the magnitude, right? So isn't that a bit of a redundant parameter?

How about:

`magnitude` : smooth only magnitude, but keep the phase
`magnitude_phase` : smooth magnitude and phase
`complex` : smooth real and imaginary part